File: Python/leetcode_53.py
from typing import List
# Given an integer array nums, find the subarray with the largest sum,
# and return its sum.


def max_sub_array(nums: List[int]) -> int:
    max_sum = float('-inf')
    current_sum = 0
    for num in nums:
        current_sum += num
        max_sum = max(max_sum, current_sum)
        current_sum = max(current_sum, 0)
    return int(max_sum)

File: JavaScript/leetcode_53.js
/* 
Given an integer array nums, find the subarray with the largest sum, and return its sum.
*/

var maxSubArray = function (nums) {
    let maxSum = -10000;
    let currentSum = 0;
    for (const num of nums) {
        currentSum += num;
        maxSum = Math.max(maxSum, currentSum);
        currentSum = Math.max(currentSum, 0);
    }
    return maxSum;
};
